lib/kernel/test/heart_SUITE.erl

```diff
@@ -21,159 +21,131 @@
+    all/0, suite/0, groups/0,
+    init_per_suite/1, end_per_suite/1
+]).
+
+%% Test case exports
+-export([
+    start/0, start/1,
+    restart/0, restart/1,
+    reboot/0, reboot/1,
+    node_start_immediately_after_crash/1,
+    node_start_soon_after_crash/1,
+    set_cmd/1, clear_cmd/1, get_cmd/1,
+    callback_api/1, options_api/1, dont_drop/1,
+    kill_pid/0, kill_pid/1,
+    heart_no_kill/0, heart_no_kill/1
+]).
+
+%% Internal exports used by peer nodes
+-export([
+    start_heart_stress/1, mangle/1,
+    suicide_by_heart/0, non_suicide_by_heart/0,
+    exhaust_atoms/0
+]).
 
--export([init_per_testcase/2, end_per_testcase/2]).
-
--export([start_heart_stress/1, mangle/1, suicide_by_heart/0, non_suicide_by_heart/0]).
-
--define(DEFAULT_TIMEOUT_SECS, 120).
-
--define(UNIQ_NODE_NAME,
-  list_to_atom(?MODULE_STRING ++ "__" ++
-               atom_to_list(?FUNCTION_NAME) ++ "_" ++
-	       integer_to_list(erlang:unique_integer([positive])))).
-
-init_per_testcase(_Func, Config) ->
-    Config.
-
-end_per_testcase(_Func, _Config) ->
-    Nodes = nodes(),
-    lists:foreach(fun(X) ->
-		NNam = list_to_atom(hd(string:tokens(atom_to_list(X),"@"))),
-		case NNam of
-		    heart_test ->
-			ct:pal(?HI_VERBOSITY, "WARNING: Killed ~p~n", [X]),
-			rpc:cast(X, erlang, halt, []);
-		    _ ->
-			ok
-		end
-	end, Nodes).
-
-%%-----------------------------------------------------------------
-%% Test suite for heart.
-%% Should be started in a CC view with:
-%% erl -sname master -rsh ctrsh
 %%-----------------------------------------------------------------
+%% Test suite for heart
 suite() ->
-    [{ct_hooks,[ts_install_cth]},
-     {timetrap,{minutes,2}}].
-
-all() -> [
-	start, restart, reboot,
-	node_start_immediately_after_crash,
-	node_start_soon_after_crash,
-	set_cmd, clear_cmd, get_cmd,
-	callback_api,
-        options_api,
-	kill_pid,
-        heart_no_kill
+    [{ct_hooks, [ts_install_cth]}, {timetrap, {minutes,2}}].
+
+all() ->
+    [{group, parallel}].
+
+groups() ->
+    [
+        {parallel, [parallel], [
+            start, restart, reboot,
+            node_start_immediately_after_crash,
+            node_start_soon_after_crash,
+            set_cmd, clear_cmd, get_cmd, callback_api,
+            options_api, kill_pid, heart_no_kill]}
     ].
 
-groups() -> 
-    [].
-
-init_per_group(_GroupName, Config) ->
-    Config.
+init_per_suite(Config) when is_list(Config) ->
+    case os:type() of
+        {win32, windows} ->
+            {skipped, "No use to run on Windows 95/98"};
+        _ ->
+            Config
+    end.
 
-end_per_group(_GroupName, Config) ->
+end_per_suite(Config) when is_list(Config) ->
     Config.
 
+%%-----------------------------------------------------------------
+%% Test suite for heart
 
-init_per_suite(Config) when is_list(Config) ->
+heart_args() ->
+    Common = ["-heart"],
     case os:type() of
-	{win32, windows} ->
-	    {skipped, "No use to run on Windows 95/98"};
-	_ ->
-	    ignore_cores:init(Config)
```

Comment:
The ignire_cores call in init_per_suite and end_per_suite needs to be there for the core ignore things to work.